### PR TITLE
Allow customisable auth path for all other auth methods

### PR DESCRIPTION
**Pull request by adammw**
Clone of https://github.com/hashicorp/vault-ruby/pull/200 upstream.

/cc @zendesk/compute @zendesk/config

### CODE REVIEW

**Review by hgokavarapuz — APPROVED** [review]

File: lib/vault/api/auth.rb [hunk 1]
```diff
@@ -29,12 +29,13 @@ class Authenticate < Request
     #
     # @param [String] new_token
     #   the new token to try to authenticate and store on the client
-    #
+    # @param [String] path (default: 'token')
+    #   The path to the auth backend to use for the login procedure.
     # @return [Secret]
-    def token(new_token)
+    def token(new_token, path = 'token')
```
Comment (grosser) [inline lib/vault/api/auth.rb]:
```suggestion
    def token(new_token, path: 'token')
```

File: lib/vault/api/auth.rb [hunk 2]
```diff
@@ -64,11 +65,12 @@ def token(new_token)
     # @param [Hash] options
     #   additional options to pass to the authentication call, such as a custom
     #   mount point
-    #
+    # @param [String] path (default: 'app-id')
+    #   The path to the auth backend to use for the login procedure.
     # @return [Secret]
-    def app_id(app_id, user_id, options = {})
+    def app_id(app_id, user_id, options = {}, path = 'app-id')
```
Comment (grosser) [inline lib/vault/api/auth.rb]: 👁 ♋️

File: lib/vault/api/auth.rb [hunk 3]
```diff
@@ -87,12 +89,13 @@ def app_id(app_id, user_id, options = {})
     # @param [String] role_id
     # @param [String] secret_id (default: nil)
     #   It is required when `bind_secret_id` is enabled for the specified role_id
-    #
+    # @param [String] path (default: 'approle')
+    #   The path to the auth backend to use for the login procedure.
     # @return [Secret]
-    def approle(role_id, secret_id=nil)
+    def approle(role_id, secret_id=nil, path='approle')
```
Comment (grosser) [inline lib/vault/api/auth.rb]: 👁 ♋️

**Review by grosser — CHANGES_REQUESTED** [review]
keyword args :D

**Comment by adammw** [comment]
I mean, the library isn't using keyword args anywhere else...

**Comment by grosser** [comment]
does not hurt to start ... at least don't put optional arguments after options hash

**Review by grosser — APPROVED** [review]

**Comment by grosser** [comment]
tests would be good ... but idk what the testing infrastructure looks like here 🤷‍♂️

**Comment by adammw** [comment]
so ldap and tls options are completely untested in the integration tests, but added tests for the other ones.

**Comment by grosser** [comment]
they all look very similar, can they get moved to a single generic auth method ?

**Review by grosser — APPROVED** [review]

**Comment by adammw** [comment]
:shurg:

don't want to deal too much more with this gem right now. this does what i need.

**Pull request merged**